tests/unit/lib/config_tests.py

```diff
@@ -239,6 +240,51 @@ def test_tupleof_invalid(self):
             parser("a, b")
 
 
+class DefaultFromEnvTests(unittest.TestCase):
+    def __init__(self, name):
+        self.unset_env_var_name = "NOT_PROVIDED"
+        self.env_var_name = "BASEPLATE_DEFAULT_VALUE"
+        self.env_var_value = "default"
+        return super().__init__(name)
+
+    def setUp(self):
+        cur_value = os.getenv(self.env_var_name)
+        self.set_env(self.env_var_name, self.env_var_value)
+        self.addCleanup(self.set_env, self.env_var_value, cur_value)
+
+        cur_value = os.getenv(self.unset_env_var_name)
+        self.set_env(self.unset_env_var_name, "")
+        self.addCleanup(self.set_env, self.unset_env_var_name, cur_value)
+
+    def set_env(self, key, value):
+        if key:
+            if value:
+                os.environ[key] = value
+            else:
+                os.environ[key] = ""
```

Comment:
[patch.dict can be used for patching the environment](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict)